apps/prairielearn/src/lib/issues.sql

```diff
@@ -0,0 +1,35 @@
+-- BLOCK open_close_all_issues_for_course
+WITH
+  updated_issues AS (
+    UPDATE issues AS i
+    SET
+      open = $open
+    WHERE
+      i.course_id = $course_id
+      AND i.course_caused
+      AND i.open IS DISTINCT FROM $open
+    RETURNING
+      i.*
```

Comment:
Is the query planner smart enough to avoid actually "instantiating" (idk exactly the right terminology) an entire issue row? Otherwise let's switch back to only returning specific columns, as issues rows can contain quite a bit of data.

Reply:
Done in c3f43c65c461a3d0b4feffb35faf0c93ae3f2a41.

Reply:
My understanding is that as of v12, postgres inlines CTEs into the main query and thus performs normal optimizations on them. That is, it no longer materializes the results of the CTE. I don't know for sure in this case, but I suspect that it is indeed smart enough to only load the columns it actually needs. See, for example, https://medium.com/alan/ctes-optimization-8c082efc47ef

Reply:
I saw you changed this, which is fine, but for future reference I think this may not be a concern.

Reply:
> See, for example, https://medium.com/alan/ctes-optimization-8c082efc47ef

If I read that correctly, that only applies to queries without side-effects. An UPDATE is certainly not in that category.

Reply:
Excellent point, I hadn't actually looked at what the CTE was doing 🤦. You are correct that CTEs with side-effects (like UPDATE) are automatically MATERIALIZED (see [the docs](https://www.postgresql.org/docs/current/queries-with.html#QUERIES-WITH-CTE-MATERIALIZATION)) so in this case I agree that it's a good idea to only return the columns we want.